### PR TITLE
AST-2287 - fix: Disable the title from elementor setting is not working as expected

### PR DESCRIPTION
**Pull request by yssambare12**
### Description

If I disable the title from the Elementor settings (individual page) it disables the title but keeps some space on the above from the Astra title.

Here I have created a short screencast for you to show you exactly how you can replicate this. Follow this: https://share.bsf.io/eDu75E6G)

### Checklist:
- [x] My code is tested
- [x] My code passes the PHPCS tests
- [x] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: inc/markup-extras.php [hunk 1]
```diff
@@ -1673,3 +1673,22 @@ function astra_skip_elementor_onboarding( $network_wide ) {
...
 
 add_action( 'activate_elementor/elementor.php', 'astra_skip_elementor_onboarding' );
+
+add_filter( 'astra_entry_header_class', 'astra_entry_header_class_custom', 1, 99 );
+/**
+ * For elementor title hide.
+ *
+ * @param bool to check if elementor is running and use elementor title hide option.
+ *
+ * @since 3.9.5
```
Comment (nathanrodrigues2111) [inline inc/markup-extras.php:1683]:
```suggestion
 * @since x.x.x
```

File: inc/markup-extras.php [hunk 2]
```diff
@@ -1673,3 +1673,22 @@ function astra_skip_elementor_onboarding( $network_wide ) {
 }
 
 add_action( 'activate_elementor/elementor.php', 'astra_skip_elementor_onboarding' );
+
+add_filter( 'astra_entry_header_class', 'astra_entry_header_class_custom', 1, 99 );
+/**
+ * For elementor title hide.
```
Comment (nathanrodrigues2111) [inline inc/markup-extras.php:1679]:
```suggestion
 * Hide elementor title.
```

File: changelog.txt [hunk 3]
```diff
@@ -20,6 +20,7 @@ v4.0.0-beta.1 (Unreleased)
 - Fix: WooCommerce - Sticky add to cart is not working on old header footer.
 - Fix: Margin applying for the last child 'li' tag inside widget section.
 - Fix: Block editor - Button background color gets overridden by core styles for the Legacy block editor setup.
+- Fix: Disable the title from elementor setting is not working as expected
```
Comment (nathanrodrigues2111) [inline changelog.txt:23]:
```suggestion
- Fix: Disable the title from elementor setting is not working as expected.
```

File: inc/markup-extras.php (hunk 1 above)
Comment (yssambare12) [reply inc/markup-extras.php]: Updated.

File: inc/markup-extras.php (hunk 2 above)
Comment (yssambare12) [reply inc/markup-extras.php]: Updated

File: changelog.txt (hunk 3 above)
Comment (yssambare12) [reply changelog.txt]: Updated.

**Review by nathanrodrigues2111 — APPROVED** [review]

File: inc/markup-extras.php [hunk 4]
```diff
@@ -1673,3 +1673,23 @@ function astra_skip_elementor_onboarding( $network_wide ) {
...
+	$post_id   = astra_get_post_id(); 
+	/** @psalm-suppress InvalidArgument */ // phpcs:ignore Generic.Commenting.DocComment.MissingShort
+	$edit_mode = get_post_meta( $post_id, '_elementor_edit_mode', true ); 
+
+	if ( $edit_mode && $edit_mode == 'builder' ) {
+		$classes[] = 'ast-header-without-markup';
+	}
+
+	return $classes;
+}
```
Comment (imnavanath) [inline inc/markup-extras.php:1695]: Get this code added in elementor compatibility file not in markup-extras.php

Comment (yssambare12) [reply inc/markup-extras.php]: Code added to the elementor compatibility file.

File: inc/compatibility/class-astra-elementor.php [hunk 5]
```diff
@@ -65,6 +65,33 @@ public function __construct() {
...
+		 * @param array $classes Array of elementor edit mode check.
+		 *
+		 * @since x.x.x
+		 */
+		function astra_entry_header_class_custom( $classes ) {
+			$post_id = astra_get_post_id(); 
+			/** @psalm-suppress InvalidArgument */ // phpcs:ignore Generic.Commenting.DocComment.MissingShort
+			$edit_mode = get_post_meta( $post_id, '_elementor_edit_mode', true ); 
+
+			if ( $edit_mode && $edit_mode == 'builder' ) {
```
Comment (imnavanath) [inline inc/compatibility/class-astra-elementor.php:90]:
```suggestion
			if ( $edit_mode && $edit_mode === 'builder' ) { 
```

**Review by imnavanath — APPROVED** [review]

**Pull request merged**